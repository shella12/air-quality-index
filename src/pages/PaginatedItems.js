import { useEffect, useState } from 'react';
import ReactPaginate from 'react-paginate';
import { PropTypes } from 'prop-types';
import { useSelector } from 'react-redux';
import CitiesCards from '../components/CitiesCards';

const PaginatedItems = (props) => {
  const { itemsPerPage, inputLocation } = props;
  const cities = useSelector((state) => state.CitiesSlice.filterCities);
  const [currentCities, setCurrentCities] = useState(null);
  const [pageCount, setPageCount] = useState(0);
  const [itemOffset, setItemOffset] = useState(0);

  useEffect(() => {
    const endOffset = itemOffset + itemsPerPage;
    setCurrentCities(cities.slice(itemOffset, endOffset));
    setPageCount(Math.ceil(cities.length / itemsPerPage));
  }, [itemOffset, itemsPerPage, cities]);

  // Invoke when user click to request another page.
  const handlePageClick = (event) => {
    const newOffset = (event.selected * itemsPerPage) % cities.length;
    setItemOffset(newOffset);
  };

  return (
    <div className="cities">
      <CitiesCards cities={currentCities} inputLocation={inputLocation} />
      <ReactPaginate
        className="pagination"
        breakLabel="..."
        nextLabel="next >"
        onPageChange={handlePageClick}
        pageRangeDisplayed={5}
        marginPagesDisplayed={2}
        pageCount={pageCount}
        previousLabel="< previous"
        renderOnZeroPageCount={null}
      />
    </div>
  );
};

PaginatedItems.propTypes = {
  itemsPerPage: PropTypes.number.isRequired,
  inputLocation: PropTypes.func.isRequired,
};

export default PaginatedItems;
