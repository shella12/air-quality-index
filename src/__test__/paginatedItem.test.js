import { render } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/configureStore';
import PaginatedItems from '../pages/PaginatedItems';

describe('testing PaginatedItems component', () => {
  it('should render the PaginatedItems component', () => {
    const paginatedItems = render(
      <Provider store={store}>
        <BrowserRouter>
          <PaginatedItems />
        </BrowserRouter>
      </Provider>,
    );
    expect(paginatedItems).toMatchSnapshot();
  });
});
