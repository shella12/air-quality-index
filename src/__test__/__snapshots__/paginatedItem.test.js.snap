// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing PaginatedItems component should render the PaginatedItems component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="cities"
      >
        <div
          class="city-list"
        />
        <ul
          aria-label="Pagination"
          class="pagination"
          role="navigation"
        >
          <li
            class="previous disabled"
          >
            <a
              aria-disabled="true"
              aria-label="Previous page"
              class=" "
              rel="prev"
              role="button"
              tabindex="-1"
            >
              &lt; previous
            </a>
          </li>
          <li
            class="next"
          >
            <a
              aria-disabled="false"
              aria-label="Next page"
              class=""
              rel="next"
              role="button"
              tabindex="0"
            >
              next &gt;
            </a>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="cities"
    >
      <div
        class="city-list"
      />
      <ul
        aria-label="Pagination"
        class="pagination"
        role="navigation"
      >
        <li
          class="previous disabled"
        >
          <a
            aria-disabled="true"
            aria-label="Previous page"
            class=" "
            rel="prev"
            role="button"
            tabindex="-1"
          >
            &lt; previous
          </a>
        </li>
        <li
          class="next"
        >
          <a
            aria-disabled="false"
            aria-label="Next page"
            class=""
            rel="next"
            role="button"
            tabindex="0"
          >
            next &gt;
          </a>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
